rcl/test/rcl/test_events.cpp

```diff
@@ -555,3 +558,130 @@ TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_pubsub_liveliness_k
   // clean up
   tear_down_publisher_subscriber();
 }
+
+/*
+ * Basic test of publisher and subscriber incompatible qos callback events.
+ * Only implemented in opensplice at the moment.
+ */
+TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_pubsub_incompatible_qos)
```

Comment:
Refactor to use gtest parameterized tests https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#value-parameterized-tests.

See for example, https://github.com/ros2/rcl/blob/master/rcl/test/rcl/test_get_actual_qos.cpp.

Reply:
Done.